### ISSUE TITLE
TypeError: this.symbolResolver.getSymbolByModule is not a function

### ISSUE THREAD

**Issue by berardo** [mgechev/ngworld#1]
Hey Minko,

Thanks for helping me on Twitter (used my company's profile, @especializa), but I still can't seem to find a way to get it working.

I tried it against several different projects but always the same output:
```
📝  Creating a program...
✅  Program created!
/usr/local/lib/node_modules/ngworld/node_modules/@angular/compiler/bundles/compiler.umd.js:23289
        return this.findSymbolDeclaration(this.symbolResolver.getSymbolByModule(moduleUrl, name, containingFile));
                                                              ^

TypeError: this.symbolResolver.getSymbolByModule is not a function
    at StaticReflector.findDeclaration (/usr/local/lib/node_modules/ngworld/node_modules/@angular/compiler/bundles/compiler.umd.js:23289:63)
    at StaticReflector.initializeConversionMap (/usr/local/lib/node_modules/ngworld/node_modules/@angular/compiler/bundles/compiler.umd.js:23507:36)
    at new StaticReflector (/usr/local/lib/node_modules/ngworld/node_modules/@angular/compiler/bundles/compiler.umd.js:23236:14)
    at ContextSymbols.init (/usr/local/lib/node_modules/ngworld/node_modules/ngast/ngast.bundle.js:2479:26)
    at new ContextSymbols (/usr/local/lib/node_modules/ngworld/node_modules/ngast/ngast.bundle.js:2340:14)
    at ProjectSymbols.getRootContext (/usr/local/lib/node_modules/ngworld/node_modules/ngast/ngast.bundle.js:2558:28)
    at Object.exports.parse (/usr/local/lib/node_modules/ngworld/parser/index.js:16:47)
    at Object.<anonymous> (/usr/local/lib/node_modules/ngworld/index.js:33:37)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
```
My current configurations:
Node: v7.9.0
NPM: 4.2.0
ngworld: 0.0.4

The latest target project config:
@angular libs: 4.1.0
typescript: 2.3.1

I've run `ng build --prod --aot` previously and served with no error.

Even on a fresh new angular-cli project that's the same outcome.

Also tried ngworld locally, running this npm script:
`"ngworld": "ngworld -p src/tsconfig.app.json"`
On terminal `npm run ngworld`

Nuking node_modules and `npm i` again doesn't seem to help.

Would you still have any idea?

**Comment by wstrinz** [comment]
getting this too. fwiw I'm trying it with an Ionic 3 project

**Comment by bnymncs** [comment]
I'm getting this error too. I've created a project with angular-cli and it works perfectly with ngrev.
I have a presentation for next week about ng-conf. I'd like to show a demo of ngworld on our current angular project to my colleagues if possible. If not, I'll use the demo. 
If you don't have time and there is anything I can help to fix it, I'll help you gladly.

**Comment by mgechev** [comment]
I haven't updated ngworld to the latest version of ngast and respectively the compiler. I will see what I can do today and keep you posted.

**Comment by mgechev** [comment]
Just updated to latest `ngast@0.0.26`. Let me know if you're still facing this problem.

Do not forget to `npm install` the dependencies of your project first. Also, make sure that you're using a global `ngworld` **not** `node_modules/.bin/ngworld` installed in the project's directory.

Finally, if it doesn't work for your own project, you can demo it with angular-seed. I'd also appreciate  if you can share a minimum demo project where I'll be able to reproduce the issue.

**Comment by bnymncoskuner** [comment]
Installing it on my PC right now, also I'll try it at work tomorrow. I'll keep you posted about how it goes.
Thanks for responding so quickly.

**Comment by bnymncoskuner** [comment]
Just tried it with a personal project which I started few months ago and it is poorly designed (I was trying to learn angular back then). 
I have only AppModule, ngworld generated a nice garden with AppModule and all of my components within it. However, it also gave me other little gardens of modules I'm not familiar with and I don't use. 
I'll try it with our application at work which is a lot better and has many modules and post the results here.
Other than that, this issue seems resolved. You can close it. Thank you.

**Comment by mgechev** [comment]
This other module might be an internal Angular one. Does it has `Ѳ` as prefix?

**Comment by bnymncoskuner** [comment]
I figured it out. I import Ng2BootstrapModule from 'ngx-bootstrap' in my AppModule and all of the other modules, generated by ngworld, are exported by Ng2BootstrapModule. Is this intentional behavior? Do you think it would get too big quickly?

**Comment by mgechev** [comment]
You can also fork the project and filter whichever modules you want.

Here's where you can [find the `formatModules` function](https://github.com/mgechev/ngworld/blob/master/parser/formatters.ts#L33-L39). You can just drop the modules you don't need.

**Comment by bnymncoskuner** [comment]
I'll try that. How can I distinguish my modules from 3rd party modules? Any tips where to start?

Also, I'd like to share the results of my trials at work. We just started this project and it has just couple of modules. Ngworld, however, captures every 3rd party modules (imported by my feature modules) and create them too. Even for a small project, it becomes a huge world and little bit laggy. 

Here are the screenshots:

I don't know how the second ss happened but when I opened the world and looked around, it became not responding for few seconds and I found myself up in the air, slowly descending.

<img width="1437" alt="screen shot 2017-05-16 at 09 31 47" src="https://cloud.githubusercontent.com/assets/1609114/26092689/013dd7e2-3a1b-11e7-837c-5bd2301e9264.png">
<img width="1425" alt="screen shot 2017-05-16 at 09 33 41" src="https://cloud.githubusercontent.com/assets/1609114/26092692/050c4868-3a1b-11e7-9f09-e992ff02c811.png">
